### PR TITLE
La 196 dipcc PATCH logic

### PR DESCRIPTION
**Pull request by Vadym-Lopatka**
[LA-196]

[LA-196]: https://selina.atlassian.net/browse/LA-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **vadym-lopatka**
You can assign the PR to them by writing `/assign @vadym-lopatka` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
@Vadym-Lopatka: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 2307f5e2de9d2a00ed926493d0f4010b7ff0e5ae | [link](http://dashboard-jx.selinadev.co.uk/namespaces/jx/pipelineruns/finance-ms-lending-api-pr-99-pr-8gmdl) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by selina-bot-user** [comment]
@Vadym-Lopatka: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 18b267943440a68ac1586fd12552fff79b462a14 | [link](http://dashboard-jx.selinadev.co.uk/namespaces/jx/pipelineruns/finance-ms-lending-api-pr-99-pr-wj8rq) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by selina-bot-user** [comment]
@Vadym-Lopatka: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 31ce6d36c56ab43abd94f183dd4ad719bde1dffc | [link](http://dashboard-jx.selinadev.co.uk/namespaces/jx/pipelineruns/finance-ms-lending-api-pr-99-pr-lkxr2) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/main/java/com/selina/lending/api/controller/CreditCommitmentsOperations.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.validation.Valid;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RequestMapping("/application")
+public interface CreditCommitmentsOperations {
+
```
Comment (bpoon-selina) [inline src/main/java/com/selina/lending/api/controller/CreditCommitmentsOperations.java:41]: Hide the api for now until we are ready for it to be used

File: src/main/java/com/selina/lending/api/controller/CreditCommitmentsOperations.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    @Operation(description = "Update an application's Credit Commitments for the given external application id")
+    @ApiResponses(value = {
+            @ApiResponse(
+                    responseCode = "200",
+                    description = "Offers based on updated Credit Commitments",
+                    content = {@Content(
+                            mediaType = APPLICATION_JSON_VALUE,
+                            schema = @Schema(implementation = QuickQuoteResponse.class))}),
```
Comment (bpoon-selina) [inline src/main/java/com/selina/lending/api/controller/CreditCommitmentsOperations.java:49]: Should not be QuickQuoteResponse here

File: src/main/java/com/selina/lending/internal/service/creaditCommitments/UpdateCreditCommitmentsServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    private final CreditCommitmentsRepository commitmentsRepository;
+
+    public UpdateCreditCommitmentsServiceImpl(MiddlewareApplicationServiceRepository applicationRepository, CreditCommitmentsRepository commitmentsRepository) {
+        this.applicationRepository = applicationRepository;
+        this.commitmentsRepository = commitmentsRepository;
+    }
+
+    @Override
+    public ApplicationResponse patchCreditCommitments(String externalId, UpdateCreditCommitmentsRequest request) {
+        var identifier = applicationRepository.getAppIdByExternalId(externalId);
```
Comment (bpoon-selina) [inline src/main/java/com/selina/lending/internal/service/creaditCommitments/UpdateCreditCommitmentsServiceImpl.java:39]: we will need to check the source account too to ensure they can run the patch request for the application

**Comment by selina-bot-user** [comment]
@Vadym-Lopatka: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 5098249641a4d992fee10d8b847c24a6b6c72839 | [link](http://dashboard-jx.selinadev.co.uk/namespaces/jx/pipelineruns/finance-ms-lending-api-pr-99-pr-l8q6j) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/main/java/com/selina/lending/internal/service/application/domain/creditCommitments/PatchCCResponse.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.selina.lending.internal.service.application.domain.creditCommitments;
+
+public class PatchCCResponse {
```
Comment (bpoon-selina) [inline src/main/java/com/selina/lending/internal/service/application/domain/creditCommitments/PatchCCResponse.java:20]: I think we should call this CreditCommitmentResponse.  The response appears to be the whole of the credit commitments response - for example, if we need to call middleware get credit commitments, the response is actually the same as the patch request

**Review by bpoon-selina — CHANGES_REQUESTED** [review]

**Comment by selina-bot-user** [comment]
@Vadym-Lopatka: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr | 5098249641a4d992fee10d8b847c24a6b6c72839 | [link](http://dashboard-jx.selinadev.co.uk/namespaces/jx/pipelineruns/finance-ms-lending-api-pr-99-pr-52rtx) | `/test pr`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: src/main/java/com/selina/lending/api/controller/CreditCommitmentsOperations.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.validation.Valid;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+//@RequestMapping("/application") TODO: hide until we will be ready to open(depends on clear MW contract)
```
Comment (bpoon-selina) [inline src/main/java/com/selina/lending/api/controller/CreditCommitmentsOperations.java:39]: you can use @Hidden annotation so you can still invoke this api, but its not visible in swagger

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-99** [here](http://ms-lending-api-pr99.selinadev.co.uk)

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[bpoon-selina](https://github.com/Selina-Finance/ms-lending-api/pull/99#issuecomment-1329179175 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)~~ [bpoon-selina]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-99** [here](http://ms-lending-api-pr99.selinadev.co.uk)

**Pull request merged**